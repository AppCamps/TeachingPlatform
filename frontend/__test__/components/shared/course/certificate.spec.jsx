import React from 'react';
import PropTypes from 'prop-types';

import { mount } from 'enzyme';

import Certificate from '../../../../components/shared/course/certificate';

const translationContext = {
  context: {
    t(trans) {
      return trans;
    },
  },
  contextTypes: PropTypes.func.isRequire,
};

describe('<Certificate/>', () => {
  const defaultProps = {
    number: 42,
    color: 'red',
  };

  it('should render isAvailable false state', () => {
    const wrapper = mount(
      <Certificate {...defaultProps} isAvailable={false} />,
      translationContext,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render isAvailable true state', () => {
    const wrapper = mount(<Certificate {...defaultProps} isAvailable />, translationContext);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render isCompleted true state', () => {
    const wrapper = mount(<Certificate {...defaultProps} isCompleted />, translationContext);

    expect(wrapper).toMatchSnapshot();
  });

  it('calls downloadCertificate with courseSchoolClass on click if isAvailable true', () => {
    const downloadCertificate = jest.fn();
    const courseSchoolClass = { certificateDownloaded: true };

    const wrapper = mount(
      <Certificate
        {...defaultProps}
        isAvailable
        courseSchoolClass={courseSchoolClass}
        downloadCertificate={downloadCertificate}
      />,
      translationContext,
    );

    wrapper.find('a[role="button"]').simulate('click');

    expect(downloadCertificate.mock.calls).toHaveLength(1);
    expect(downloadCertificate.mock.calls[0]).toEqual([courseSchoolClass]);
  });
});
