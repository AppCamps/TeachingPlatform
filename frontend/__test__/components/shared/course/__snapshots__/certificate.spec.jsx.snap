// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Certificate/> should render isAvailable false state 1`] = `
<Certificate
  certificateUrl={null}
  color="red"
  courseSchoolClass={null}
  downloadCertificate={[Function]}
  isAvailable={false}
  isCompleted={false}
  isProgressIndicator={false}
  number={42}
>
  <div
    className="certificate"
  >
    <span
      className="help"
    >
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        title="Click to get help"
        to="/help/certificates"
      >
        <a
          onClick={[Function]}
          style={Object {}}
          title="Click to get help"
        >
          <HelpIcon>
            <FaIcon
              className="helpIcon"
              icon="question"
            >
              <i
                aria-hidden="true"
                className="fa fa-question helpIcon"
              />
            </FaIcon>
          </HelpIcon>
        </a>
      </Link>
    </span>
    <div
      className="disabled"
      style={
        Object {
          "borderColor": "colorGray",
          "certColor": "colorGray",
        }
      }
    >
      <div
        className="numberContainer"
        style={
          Object {
            "borderColor": "colorGray",
          }
        }
      >
        <Number
          color="colorGray"
          invert={null}
          number={42}
        >
          <span
            className="number"
            style={
              Object {
                "borderColor": "colorGray",
                "color": "colorGray",
              }
            }
          >
            42
          </span>
        </Number>
      </div>
      <div
        className="certificateTitle"
      >
        <span>
          <FaIcon
            className={null}
            icon="lock"
          >
            <i
              aria-hidden="true"
              className="fa fa-lock"
            />
          </FaIcon>
           
        </span>
        Certificates
      </div>
      <div
        className="info"
      >
        Available when all lessons are completed.
      </div>
    </div>
  </div>
</Certificate>
`;

exports[`<Certificate/> should render isAvailable true state 1`] = `
<Certificate
  certificateUrl={null}
  color="red"
  courseSchoolClass={null}
  downloadCertificate={[Function]}
  isAvailable={true}
  isCompleted={false}
  isProgressIndicator={false}
  number={42}
>
  <div
    className="certificate"
  >
    <span
      className="help"
    >
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        title="Click to get help"
        to="/help/certificates"
      >
        <a
          onClick={[Function]}
          style={Object {}}
          title="Click to get help"
        >
          <HelpIcon>
            <FaIcon
              className="helpIcon"
              icon="question"
            >
              <i
                aria-hidden="true"
                className="fa fa-question helpIcon"
              />
            </FaIcon>
          </HelpIcon>
        </a>
      </Link>
    </span>
    <a
      onClick={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className=""
        style={
          Object {
            "borderColor": "red",
            "certColor": "red",
          }
        }
      >
        <div
          className="numberContainer"
          style={
            Object {
              "borderColor": "red",
            }
          }
        >
          <Number
            color="red"
            invert={null}
            number={42}
          >
            <span
              className="number"
              style={
                Object {
                  "borderColor": "red",
                  "color": "red",
                }
              }
            >
              42
            </span>
          </Number>
        </div>
        <div
          className="certificateTitle"
        >
          Certificates
        </div>
        <div
          className="info"
        >
          Click here to download your template.
        </div>
      </div>
    </a>
  </div>
</Certificate>
`;

exports[`<Certificate/> should render isCompleted true state 1`] = `
<Certificate
  certificateUrl={null}
  color="red"
  courseSchoolClass={null}
  downloadCertificate={[Function]}
  isAvailable={false}
  isCompleted={true}
  isProgressIndicator={false}
  number={42}
>
  <div
    className="certificate"
  >
    <span
      className="help"
    >
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        title="Click to get help"
        to="/help/certificates"
      >
        <a
          onClick={[Function]}
          style={Object {}}
          title="Click to get help"
        >
          <HelpIcon>
            <FaIcon
              className="helpIcon"
              icon="question"
            >
              <i
                aria-hidden="true"
                className="fa fa-question helpIcon"
              />
            </FaIcon>
          </HelpIcon>
        </a>
      </Link>
    </span>
    <div
      className="disabled"
      style={
        Object {
          "borderColor": "colorGray",
          "certColor": "colorGray",
        }
      }
    >
      <div
        className="numberContainer"
        style={
          Object {
            "borderColor": "colorGray",
          }
        }
      >
        <Number
          color="colorGray"
          invert={null}
          number={42}
        >
          <span
            className="number"
            style={
              Object {
                "borderColor": "colorGray",
                "color": "colorGray",
              }
            }
          >
            42
          </span>
        </Number>
      </div>
      <div
        className="certificateTitle"
      >
        <span>
          <FaIcon
            className={null}
            icon="lock"
          >
            <i
              aria-hidden="true"
              className="fa fa-lock"
            />
          </FaIcon>
           
        </span>
        Certificates
      </div>
      <div
        className="info"
      >
        Available when all lessons are completed.
      </div>
    </div>
  </div>
</Certificate>
`;
